crypto/stark-verifier-ethereum/contracts/merkle.sol

```diff
@@ -0,0 +1,122 @@
+pragma solidity ^0.6.4;
+
+import './hashing.sol';
+import './ring_buffer.sol';
+
+contract MerkleVerifier is Hashing, RingBuffer {
```

Comment:
Solidity has something like `use RingBuffer for IndexRingBuffer` to allow member function syntax. This would be a good place to use it.